packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -166,7 +166,10 @@ class BuildWebCommand extends BuildSubCommand {
     }
     final String? baseHref = stringArg('base-href');
     if (baseHref != null && !(baseHref.startsWith('/') && baseHref.endsWith('/'))) {
-      throwToolExit('base-href should start and end with /');
+      throwToolExit(
+        'Received a --base-href value of "$baseHref"\n'
+        '--base-href should start and end with /',
+      );
```

Comment:
turbo self-nit: these lines should probably be in reverse order to follow the typical convention of the error being first with details following afterward

but it's fine